swarm/network/stream/messages.go

```diff
@@ -75,13 +75,13 @@ type RequestSubscriptionMsg struct {
 }
 
 func (p *Peer) handleRequestSubscription(ctx context.Context, req *RequestSubscriptionMsg) (err error) {
-	log.Debug(fmt.Sprintf("handleRequestSubscription: streamer %s to subscribe to %s with stream %s", p.streamer.addr, p.ID(), req.Stream))
-	if err = p.streamer.Subscribe(p.ID(), req.Stream, req.History, req.Priority); err != nil {
+	log.Debug(fmt.Sprintf("handleRequestSubscription: streamer %s to subscribe to %s with stream %s", p.streamer.addr, p.bzzPeer.ID(), req.Stream))
+	if err = p.streamer.Subscribe(p.bzzPeer.ID(), req.Stream, req.History, req.Priority); err != nil {
 		// The error will be sent as a subscribe error message
 		// and will not be returned as it will prevent any new message
 		// exchange between peers over p2p. Instead, error will be returned
 		// only if there is one from sending subscribe error message.
-		err = p.Send(ctx, SubscribeErrorMsg{
+		err = p.bzzPeer.Send(ctx, SubscribeErrorMsg{
```

Comment:
Why do we need to change this as well? Whether we call `Send` on `p` or `bzzPeer` is the same, no?